plugins/base/src/test/kotlin/parsers/JavadocParserTest.kt

```diff
@@ -349,4 +349,53 @@ class JavadocParserTest : BaseAbstractTest() {
             }
         }
     }
+    
+    @Test
+    fun `header tags are handled properly`() {
+        val source = """
+            |/src/main/kotlin/test/Test.java
+            |package example
+            |
+            | /**
+            | * An example of using the header tags
+            | * <h1>A header</h1>
+            | * <h2>A second level header</h2>
+            | * <h3>A third level header</h3>
+            | */
+            | public class Test  {}
+            """.trimIndent()
+        testInline(
+            source,
+            configuration,
+        ) {
+            documentablesCreationStage = { modules ->
+                val docs = modules.first().packages.first().classlikes.single().documentation.first().value
+                val root = docs.children.first().root
+
+                kotlin.test.assertEquals(
+                    listOf(
+                        P(children = listOf(
+                            Text("An example of using the header tags "),
+                            H1(
```

Comment:
`H1`, `H2`, `H3` are not the children of `P`

Reply:
It looks like the test is passing finally 😅 (on Windows) however there is an unrelated build failure on ubuntu. 

```
> Could not resolve all files for configuration ':docs:classpath'.
   > Could not resolve org.ajoberstar.grgit:grgit-core:latest.release.
     Required by:
         project :docs > ru.vyarus.mkdocs:ru.vyarus.mkdocs.gradle.plugin:2.1.1 > ru.vyarus:gradle-mkdocs-plugin:2.1.1 > org.ajoberstar:gradle-git-publish:3.0.0
      > No matching variant of org.ajoberstar.grgit:grgit-core:5.0.0-beta.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.3' but:
          - Variant 'apiElements' capability org.ajoberstar.grgit:grgit-core:5.0.0-beta.1 declares a library, packaged as a jar, and its dependencies declared externally:
```
Can you force a re-run?

Reply:
FYI: ran it 3 times just now, didn't help. Tests pass locally, `docs` project is configurable, everything looks good, so not sure what the problem is. 

One recent change that could potentially be to blame - I updated `com.gradle.plugin-publish` to the latest version in #2318, but tests ran OK at the time (for that and following PRs).

Maybe the agents the tests are running on are experiencing some issues, probably need to re-run it again after some time

Reply:
Can you this rebase onto master to pick up a3baeaded708f2cd4d174f1d412c25d88b14b9a5?

Reply:
Rebased